tsdb/head.go

```diff
@@ -1058,6 +1065,7 @@ func (a *headAppender) Commit() error {
 
 		if !ok {
 			total--
+			discarded++
```

Comment:
We can get rid of `total` variable and use `len(a.samples)-discarded` when adding the metric, looks cleaner.